task-generator/remote/main.go

```diff
@@ -103,12 +103,13 @@ func convertToSsh(task *tektonapi.Task, taskVersion string) {
 	// to prevent them from getting garbage collected before generating the image index.
 	// We can simplify this process, preventing the need for users to manually specify
 	// the image by auto-appending the architecture from the PLATFORM parameter. For
-	// example, this will append -arm64 if PLATFORM is linux/arm64 if not present. Since
-	// we cannot modify the parameter itself, this replacement needs to happen in any task
+	// example, this will append -arm64 if PLATFORM is linux/arm64 if not present. If PLATFORM
+	// does not contain a "/" then the entire platform will be appended to IMAGE (i.e. "localhost").
+	// Since we cannot modify the parameter itself, this replacement needs to happen in any task
 	// step where the IMAGE parameter is used.
 	// If a user defines the IMAGE parameter with an -arm64 suffix, the arm64 suffix will
 	// not be appended again based on the PLATFORM.
-	adjustRemoteImage := `if [[ "${IMAGE##*-}" != "${PLATFORM##*/}" ]]; then
+	adjustRemoteImage := `if [ -n "${APPEND_PLATFORM_ARCH}" ] &&  [ "${IMAGE##*-}" != "${PLATFORM##*/}" ]; then
```

Comment:
One potential concern with #1185 is that the suffix will always be added if the remote task is used. Now that we are changing the remote task to be a general-purpose task (i.e. being able to run locally as well), I modified the behavior so that this task _can_ be used as the only build task in a single-arch mode without affecting the image tag used.